dlc/src/main/scala/org/bitcoins/dlc/BinaryOutcomeDLCWithSelf.scala

```diff
@@ -0,0 +1,306 @@
+    val toRemote: TransactionOutput =
+      TransactionOutput(remotePayout - feeSoFar,
+                        P2PKHScriptPubKey(cetRemotePrivKey.publicKey))
+
+    val outputs: Vector[TransactionOutput] = Vector(toLocal, toRemote)
+    val txBuilderF =
+      BitcoinTxBuilder(outputs,
+                       Vector(fundingSpendingInfo),
+                       feeRate,
+                       changeSPK,
+                       network)
+
+    txBuilderF.flatMap(subtractFeeAndSign)
+  }
+
+  def createCETRemote(
+      sigPubKey: ECPublicKey,
+      fundingSpendingInfo: BitcoinUTXOSpendingInfo,
+      localPayout: CurrencyUnit,
+      remotePayout: CurrencyUnit): Future[Transaction] = {
+
+    val multiSig = MultiSignatureScriptPubKey(
+      requiredSigs = 2,
+      pubKeys = Vector(cetRemotePrivKey.publicKey, sigPubKey))
+    val timeoutSPK = CLTVScriptPubKey(
+      locktime = ScriptNumber(timeout),
+      scriptPubKey = P2PKHScriptPubKey(cetLocalPrivKey.publicKey))
+
+    val toLocalSPK = MultiSignatureWithTimeoutScriptPubKey(multiSig, timeoutSPK)
+
+    val toLocal: TransactionOutput =
+      TransactionOutput(remotePayout, toLocalSPK)
+    val feeSoFar = totalInput - fundingSpendingInfo.output.value
+    val toRemote: TransactionOutput =
+      TransactionOutput(localPayout - feeSoFar,
+                        P2PKHScriptPubKey(cetLocalPrivKey.publicKey))
+
+    val outputs: Vector[TransactionOutput] = Vector(toLocal, toRemote)
+    val txBuilderF =
+      BitcoinTxBuilder(outputs,
+                       Vector(fundingSpendingInfo),
+                       feeRate,
+                       changeSPK,
+                       network)
+
+    txBuilderF.flatMap(subtractFeeAndSign)
+  }
+
+  def createCETWinLocal(
+      fundingSpendingInfo: BitcoinUTXOSpendingInfo): Future[Transaction] = {
+    createCETLocal(
+      sigPubKey = sigPubKeyWin,
+      fundingSpendingInfo = fundingSpendingInfo,
+      localPayout = localWinPayout,
+      remotePayout = remoteWinPayout
+    )
+  }
+
+  def createCETLoseLocal(
+      fundingSpendingInfo: BitcoinUTXOSpendingInfo): Future[Transaction] = {
+    createCETLocal(
+      sigPubKey = sigPubKeyLose,
+      fundingSpendingInfo = fundingSpendingInfo,
+      localPayout = localLosePayout,
+      remotePayout = remoteLosePayout
+    )
+  }
+
+  def createCETWinRemote(
+      fundingSpendingInfo: BitcoinUTXOSpendingInfo): Future[Transaction] = {
+    createCETRemote(
+      sigPubKey = sigPubKeyWin,
+      fundingSpendingInfo = fundingSpendingInfo,
+      localPayout = localWinPayout,
+      remotePayout = remoteWinPayout
+    )
+  }
+
+  def createCETLoseRemote(
+      fundingSpendingInfo: BitcoinUTXOSpendingInfo): Future[Transaction] = {
+    createCETRemote(
+      sigPubKey = sigPubKeyLose,
+      fundingSpendingInfo = fundingSpendingInfo,
+      localPayout = localLosePayout,
+      remotePayout = remoteLosePayout
+    )
+  }
+
+  def executeDLC(oracleSigF: Future[SchnorrDigitalSignature]): Future[
```

Comment:
How can we tell what close case is happening based on this signature? Let's actually break things up into the different close cases to make this more understandable. 

Comments are great! :-)